src/app/qgsprojectproperties.cpp

```diff
@@ -2680,6 +2702,76 @@ void QgsProjectProperties::removeStyleDatabase()
   delete mListStyleDatabases->takeItem( currentRow );
 }
 
+#if QT_VERSION >= QT_VERSION_CHECK(6, 8, 0)
+
+void QgsProjectProperties::addIccProfile()
+{
+  const QString iccProfileFilePath = QFileDialog::getOpenFileName(
+                                       this,
+                                       tr( "Load ICC Profile" ),
+                                       QDir::homePath(),
+                                       tr( "ICC Profile" ) + QStringLiteral( " (*.icc)" ) );
+
+  addIccProfile( iccProfileFilePath );
+}
+
+void QgsProjectProperties::addIccProfile( const QString &iccProfileFilePath )
+{
+  if ( iccProfileFilePath.isEmpty() )
+    return;
+
+  QString errorMsg;
+  QColorSpace colorSpace = QgsColorUtils::iccProfile( iccProfileFilePath, errorMsg );
+  if ( !colorSpace.isValid() )
+  {
+    QMessageBox::warning( this, tr( "Load ICC Profile" ), errorMsg );
+    return;
+  }
+
+  mColorSpace = colorSpace;
+  updateColorSpaceWidgets();
+}
+
+void QgsProjectProperties::removeIccProfile()
+{
+  mColorSpace = QColorSpace();
+  updateColorSpaceWidgets();
+}
+
+void QgsProjectProperties::saveIccProfile()
+{
+  QString fileName = QFileDialog::getSaveFileName( this, tr( "Save ICC profile" ), QDir::homePath(),
+                     tr( "ICC profile files (*.icc *.ICC)" ) );
+
+  if ( fileName.isEmpty() )
+    return;
+
+  const QString error = QgsColorUtils::saveIccProfile( mColorSpace, fileName );
```

Comment:
We need this to ensure the extension is always present on all environments (eg gnome won't add it by default):

```suggestion
  fileName = QgsFileUtils::ensureFileNameHasExtension( fileName, { QStringLiteral( "icc" ) } );
  const QString error = QgsColorUtils::saveIccProfile( mColorSpace, fileName );
```